Vapor tests got broken by AHC tracing change

### Describe the issue

Vapor tests got broken by AHC tracing change swift-server/async-http-client#861

### Vapor version

`main`

### Operating system and version

Ubuntu Linux 22.04

### Swift version

Swift 6.1

### Steps to reproduce

Clone Vapor. Change the AHC dependency to depend on the `main` branch of AHC.

### Outcome

```
       > Test Suite 'Selected tests' started at 2025-10-09 12:26:47.043
       > Test Suite 'MiddlewareTests' started at 2025-10-09 12:26:47.061
       > Test Case 'MiddlewareTests.testTracingMiddleware' started at 2025-10-09 12:26:47.061
       > 2025-10-09T12:26:47+0000 notice codes.vapor.application: [Vapor] Server started on http://127.0.0.1:43177
       > 2025-10-09T12:26:47+0000 info codes.vapor.application: request-id=8393C40A-0523-42FE-9DE8-5675FCA8CB81 [Vapor] GET /testTracing
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:253: error: MiddlewareTests.testTracingMiddleware : XCTAssertEqual failed: ("GET") is not equal to ("GET /testTracing") - 
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:255: error: MiddlewareTests.testTracingMiddleware : XCTAssertEqual failed: ("nil") is not equal to ("Optional(Tracing.SpanAttribute.string("GET"))") - 
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:256: error: MiddlewareTests.testTracingMiddleware : XCTAssertEqual failed: ("nil") is not equal to ("Optional(Tracing.SpanAttribute.string("/testTracing"))") - 
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:259: error: MiddlewareTests.testTracingMiddleware : XCTAssertEqual failed: ("nil") is not equal to ("Optional(Tracing.SpanAttribute.string("/testTracing"))") - 
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:260: error: MiddlewareTests.testTracingMiddleware : XCTAssertEqual failed: ("nil") is not equal to ("Optional(Tracing.SpanAttribute.string("http"))") - 
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:261: error: MiddlewareTests.testTracingMiddleware : XCTAssertEqual failed: ("nil") is not equal to ("Optional(Tracing.SpanAttribute.string("127.0.0.1"))") - 
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:262: error: MiddlewareTests.testTracingMiddleware : XCTAssertEqual failed: ("nil") is not equal to ("Optional(Tracing.SpanAttribute.int64(43177))") - 
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:263: error: MiddlewareTests.testTracingMiddleware : XCTAssertEqual failed: ("nil") is not equal to ("Optional(Tracing.SpanAttribute.string("foo=bar"))") - 
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:265: error: MiddlewareTests.testTracingMiddleware : XCTAssertEqual failed: ("nil") is not equal to ("Optional(Tracing.SpanAttribute.string("127.0.0.1"))") - 
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:266: error: MiddlewareTests.testTracingMiddleware : XCTAssertEqual failed: ("nil") is not equal to ("Optional(Tracing.SpanAttribute.string("127.0.0.1"))") - 
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:267: error: MiddlewareTests.testTracingMiddleware : XCTAssertNotNil failed - 
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:268: error: MiddlewareTests.testTracingMiddleware : XCTAssertEqual failed: ("nil") is not equal to ("Optional(Tracing.SpanAttribute.string("1.1"))") - 
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:269: error: MiddlewareTests.testTracingMiddleware : XCTAssertEqual failed: ("nil") is not equal to ("Optional(Tracing.SpanAttribute.string("test"))") - 
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:271: error: MiddlewareTests.testTracingMiddleware : XCTAssertEqual failed: ("nil") is not equal to ("Optional(Tracing.SpanAttribute.string("custom"))") - 
       > /workspace/session/repos/vapor/Tests/VaporTests/MiddlewareTests.swift:273: error: MiddlewareTests.testTracingMiddleware : XCTAssertEqual failed: ("nil") is not equal to ("Optional(Tracing.SpanAttribute.int64(200))") - 
       > Test Case 'MiddlewareTests.testTracingMiddleware' failed (0.051 seconds)
       > Test Suite 'MiddlewareTests' failed at 2025-10-09 12:26:47.113
       > 	 Executed 1 test, with 15 failures (0 unexpected) in 0.051 (0.051) seconds
       > Test Suite 'Selected tests' failed at 2025-10-09 12:26:47.113
       > 	 Executed 1 test, with 15 failures (0 unexpected) in 0.051 (0.051) seconds
```

### Additional notes

_No response_

--- a/Tests/VaporTests/MiddlewareTests.swift
+++ b/Tests/VaporTests/MiddlewareTests.swift
@@ -249,8 +249,7 @@ final class MiddlewareTests: XCTestCase {
         XCTAssertEqual(response.status, .ok)
         XCTAssertEqual(response.body?.string, "done")
 
-        let span = try XCTUnwrap(tracer.spans.first)
-        XCTAssertEqual(span.operationName, "GET /testTracing")
+        let span = try XCTUnwrap(tracer.spans.first(where: { $0.operationName == "GET /testTracing" }))
         
         XCTAssertEqual(span.attributes["http.request.method"]?.toSpanAttribute(), "GET")
         XCTAssertEqual(span.attributes["url.path"]?.toSpanAttribute(), "/testTracing")

